### DISCUSSION TITLE
Invalid host parameter (nil or ""). Should be a domain name or IP address string.

### DISCUSSION THREAD

**Discussion by ixiao** [rabbitmq/rabbitmq-website#1359] [General]
```
- (instancetype)initWithDelegate:(id<RMQConnectionDelegate>)delegate {
    return [self initWithUri:@"amqps://admin:p@ssw0rd3344@b-99587d00d-df55-4400-8658-8a5c5c6e23ba.mq.us-east-1.amazonaws.com:5671/" delegate:delegate];
}
```
ViewController.m
```
    RMQConnection *conn = [[RMQConnection alloc] initWithDelegate:[RMQConnectionDelegateLogger new]];
    [conn start];

    id<RMQChannel> ch = [conn createChannel];

    RMQQueue *q = [ch queue:@"adduser" options:RMQQueueDeclareDurable];

    NSData *msgData = [msg dataUsingEncoding:NSUTF8StringEncoding];
    
    [ch.defaultExchange publish:msgData routingKey:q.name persistent:YES];
    
    NSLog(@"Sent %@", msg);
    [NSThread sleepForTimeInterval:2.0f];
    
    [conn close];
```

Get Error:
```
Received connection: <RMQConnection: 0x2839d05a0> failedToConnectWithError: Error Domain=GCDAsyncSocketErrorDomain Code=2 "Invalid host parameter (nil or ""). Should be a domain name or IP address string." UserInfo={NSLocalizedDescription=Invalid host parameter (nil or ""). Should be a domain name or IP address string.}
```

I found every internet question & answer, I don't know why, android is working... 
thank u

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Reply by ixiao** [reply]
my answer : ios client can not recognize my serve name `ssw0rd3344@b-99587d00d-df55-4400-8658-8a5c5c6e23ba.mq` 
thank you ~

**Comment by michaelklishin** [comment]
Your connection does not provide a hostname to connect to. Tutorial 1 uses `initWithUri:delegate:`, [client integration tests](https://github.com/rabbitmq/rabbitmq-objc-client/tree/master/RMQClientIntegrationTests) use a bunch of other selectors.

I will delete `initWithDelegate:` from the tutorial copy since it does not contribute much compared to some other selectors.
